lib/logflare_web/controllers/api/team_controller.ex

```diff
@@ -0,0 +1,51 @@
+defmodule LogflareWeb.Api.TeamController do
+  use LogflareWeb, :controller
+  alias Logflare.Teams
+  action_fallback LogflareWeb.Api.FallbackController
+
+  def index(%{assigns: %{user: user}} = conn, _) do
+    teams = Teams.list_for_user(user)
+    json(conn, teams)
+  end
+
+  def show(%{assigns: %{user: user}} = conn, %{"token" => token}) do
+    with team when not is_nil(team) <- Teams.get(user, token),
```

Comment:
should handle situation when it is nil

Reply:
it is handled, it bubbles up to the action_fallback and returns 404